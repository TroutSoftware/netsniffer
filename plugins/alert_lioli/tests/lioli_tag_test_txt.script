# Inspectors and spells are in place to attribute the correct flow
pcap testdata/google_http.pcap
cmp output.txt testdata/lioli_tag_test_txt.expected.txt 

-- cfg.lua --

logger_file = { file_name = 'output.txt',
                serializer = 'serializer_txt' }

alert_lioli = { logger = 'logger_file',
                testmode = true }

stream = {}
stream_tcp = {}
stream_udp = {}
http_inspect = {}

wizard = {
    spells = { { service = 'http', proto = 'tcp', to_server = {'GET'}, to_client = {'HTTP/'} } }
}

binder = {
    { when = { service = 'http' }, use = { type = 'http_inspect' } },
    { use = { type = 'wizard' } }
}

ips = {
  include = 'lua.rules'
}

-- lua.rules --

alert ip any any -> any any (
  msg:"This is a log of an http header";

  http_header: field host;
  lioli_bind: $.host;
  content:"google";

  lioli_tag: $.tag "Tag test";

  lioli_tag: $.tag.nested "Nested ";
  lioli_tag: $.tag.nested "test";

  lioli_tag: $.principal.append "Append test";
  
)
