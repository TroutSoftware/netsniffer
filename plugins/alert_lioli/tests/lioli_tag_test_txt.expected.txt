vvvvvvvvvvvvvvvvvvvvvvvv
$: 1970-01-01T00:00:00.000000000Z"This is a log of an http header"http209.85.202.100:80google.com10.67.21.59:48872Append testTag testNested test
-timestamp: 1970-01-01T00:00:00.000000000Z
-alert: "This is a log of an http header"
-protocol: http
-endpoint: 209.85.202.100:80
--addr: 209.85.202.100:80
---ip: 209.85.202.100
---port: 80
-host: google.com
-principal: 10.67.21.59:48872Append test
--addr: 10.67.21.59:48872
---ip: 10.67.21.59
---port: 48872
--append: Append test
-tag: Tag testNested test
--nested: Nested test
^^^^^^^^^^^^^^^^^^^^^^^^
vvvvvvvvvvvvvvvvvvvvvvvv
$: 1970-01-01T00:00:00.000000000Z"This is a log of an http header"http209.85.202.100:80google.com10.67.21.59:48872Append testTag testNested test
-timestamp: 1970-01-01T00:00:00.000000000Z
-log: "This is a log of an http header"
-protocol: http
-endpoint: 209.85.202.100:80
--addr: 209.85.202.100:80
---ip: 209.85.202.100
---port: 80
-host: google.com
-principal: 10.67.21.59:48872Append test
--addr: 10.67.21.59:48872
---ip: 10.67.21.59
---port: 48872
--append: Append test
-tag: Tag testNested test
--nested: Nested test
^^^^^^^^^^^^^^^^^^^^^^^^
vvvvvvvvvvvvvvvvvvvvvvvv
$: 1970-01-01T00:00:00.000000000Z"This is a log of an http header"http172.253.116.147:80www.google.com10.67.21.59:55904Append testTag testNested test
-timestamp: 1970-01-01T00:00:00.000000000Z
-alert: "This is a log of an http header"
-protocol: http
-endpoint: 172.253.116.147:80
--addr: 172.253.116.147:80
---ip: 172.253.116.147
---port: 80
-host: www.google.com
-principal: 10.67.21.59:55904Append test
--addr: 10.67.21.59:55904
---ip: 10.67.21.59
---port: 55904
--append: Append test
-tag: Tag testNested test
--nested: Nested test
^^^^^^^^^^^^^^^^^^^^^^^^
vvvvvvvvvvvvvvvvvvvvvvvv
$: 1970-01-01T00:00:00.000000000Z"This is a log of an http header"http172.253.116.147:80www.google.com10.67.21.59:55904Append testTag testNested test
-timestamp: 1970-01-01T00:00:00.000000000Z
-log: "This is a log of an http header"
-protocol: http
-endpoint: 172.253.116.147:80
--addr: 172.253.116.147:80
---ip: 172.253.116.147
---port: 80
-host: www.google.com
-principal: 10.67.21.59:55904Append test
--addr: 10.67.21.59:55904
---ip: 10.67.21.59
---port: 55904
--append: Append test
-tag: Tag testNested test
--nested: Nested test
^^^^^^^^^^^^^^^^^^^^^^^^
------------------------
